api/src/test/java/org/motechproject/nms/api/osgi/MobileAcademyControllerBundleIT.java

```diff
@@ -79,12 +85,27 @@ public void testSetValidBookmark() throws IOException, InterruptedException {
     }
 
     @Test
-    public void testGetCourse() throws IOException, InterruptedException {
+    public void testGetCourseNotPresent() throws IOException, InterruptedException {
```

Comment:
I went back and forth on this one. The MA service *has* to be initialized
or bootstrapped with the one and only course for it and it is always
guaranteed to exist...hence the 500.

I think this would be a 404 in a majority of implementations (if not all)
but NMS is special :(
On May 7, 2015 8:37 AM, "Frank Huster" <notifications@github.com> wrote:

> In
> api/src/test/java/org/motechproject/nms/api/osgi/MobileAcademyControllerBundleIT.java
> <https://github.com/motech-implementations/mim/pull/222#discussion_r29864056>
> :
>
> > @@ -79,12 +85,27 @@ public void testSetValidBookmark() throws IOException, InterruptedException {
> >      }
> >
> >      @Test
> > -    public void testGetCourse() throws IOException, InterruptedException {
> > +    public void testGetCourseNotPresent() throws IOException, InterruptedException {
>
> Shouldn't something being not present return a 404 instead of a 500?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/222/files#r29864056>.
>